src/composition/supergraph/config/resolver/mod.rs

```diff
@@ -117,7 +120,30 @@ impl SupergraphConfigResolver<state::LoadRemoteSubgraphs> {
                 .call(FetchRemoteSubgraphsRequest::new(graph_ref.clone()))
                 .await
                 .map_err(|err| {
-                    LoadRemoteSubgraphsError::FetchRemoteSubgraphsError(Box::new(err))
+
+                    // Q1: Academic: Why choose map_err here instead of From? I think I could use
+                    // the From trait to cast the error raised by the try operator (?) into
+                    // whatever I want. I'm chosing to use map_err here becuase it is how the rest
+                    // of this code is structure, but curious about the idiomatic approach here.
+
+
+                    // Q2: Practical: I want to write a match statement that has an arm that matches
+                    // the `err` has an "Invalid credentials provided" message. I've copied what
+                    // `err` logs to the console below.
+
+                    // eprintln!("Error => \n {:?}", err);
+
+                    //Error =>
+                    //Service { source: PartialError { data: ResponseData { variant: None }, errors: [Error { message: "406: Not Acceptable", locations: None, path: None, extensions: Some({"response": Object {"body": Object {"errors": Array [Object {"message": String("Invalid credentials provided")}]}, "status": Number(406), "statusText": String("Not Acceptable"), "url": String("http://engine-graphql.apollo-default.svc.prod.apollo-internal:8081/api/graphql")}, "code": String("INTERNAL_SERVER_ERROR")}) }] }, endpoint_kind: ApolloStudio }
+                    //LoadRemoteSubgraphsError::FetchRemoteSubgraphsError(Box::new(err))
```

Comment:
This is intended as an example. This is a simple question to an experience dev, but I don't know how to navigate the entity return in `err` so that I could match on a deeply nested property in my match statemement.